activesupport/lib/active_support/core_ext/marshal.rb

```diff
@@ -6,6 +6,9 @@ def load_with_autoloading(source)
       load_without_autoloading(source)
     rescue ArgumentError, NameError => exc
       if exc.message.match(%r|undefined class/module (.+)|)
+        processed_names ||= {}
+        raise exc if processed_names[$1]
```

Comment:
What about to use [named capture](http://stackoverflow.com/a/18825787/319233) instead of `$1`?